src/asn1.ml

```diff
@@ -615,11 +667,263 @@ struct
+              | _ ->
+                  raise (Failure "Invalid LENGTH field in TLV encoded CVC data")
+            end
+      | Value length ->
+          let is_rec =
+            match state with
+              | None -> false
+              | Some (_, (_, x)) -> x
+          in
+          let acc =
+            if is_rec
+            then
+              `Value (tokenize ~acc:[] bytes i (i + length) None Init) :: acc
+            else
+              let bytes' =
+                Cstruct.sub bytes i length
+              in
+              `Bytes bytes' :: acc
+          in
+          (if length + i >= Cstruct.len bytes then List.rev acc else (tokenize[@tailcall]) ~acc bytes (i + length) lim None Init)
+    in
+    let tokens = tokenize ~acc:[] bytes 0 (Cstruct.len bytes) None Init in
+    let rec parse = function
+      | `Type (_, (`PUBLIC_KEY, _)) :: `Length _ :: `Value ls :: _ ->
+          parse ls
+      | `Type (_, (`OID, _)) :: `Length _ :: `Bytes bytes :: tl ->
+          let bytes =
+            let prefix =
+              Printf.sprintf "\006%c" (Char.chr (Cstruct.len bytes))
+              |> Cstruct.of_string
+            in
+            Cstruct.append prefix bytes
+          in
+          `Oid (decode_oid bytes) :: parse tl
+      | `Type (_, (`MODULUS, _)) :: `Length _ :: `Bytes bytes :: tl ->
+          `Modulus (atoz_bigendian bytes) :: parse tl
+      | `Type (0x82, ((*`EXPONENT*) _ , _)) :: `Length _ :: `Bytes bytes :: tl ->
+          `Exponent (atoz_bigendian bytes) :: parse tl
+      | `Type (_, (`COEFFICIENT_B, _)) :: `Length _ :: `Bytes bytes :: tl ->
+          `Coefficient_b (atoz_bigendian bytes) :: parse tl
+      | `Type (_, (`BASE_POINT_G, _)) :: `Length _ :: `Bytes bytes :: tl ->
+          `Base_point_g (atoz_bigendian bytes) :: parse tl
+      | `Type (_, (`BASE_POINT_R_ORDER, _)) :: `Length _ :: `Bytes bytes :: tl ->
+          `Base_point_r_order (atoz_bigendian bytes) :: parse tl
+      | `Type (_, (`PUBLIC_POINT_Y, _)) :: `Length _ :: `Bytes bytes :: tl ->
+          `Public_point_y (atoz_bigendian bytes) :: parse tl
+      | `Type (_, (`COFACTOR_F, _)) :: `Length _ :: `Bytes bytes :: tl ->
+          `Cofactor_f (atoz_bigendian bytes) :: parse tl
+      | [] ->
+          []
+      | `Type (t, _) :: tl ->
+          let () = Printf.printf "`Type (0x%x, _) :: tl found\n" t in
+          parse tl
+      | `Length _ :: tl ->
+          let () = Printf.printf "`Length _ :: tl found\n" in
+          parse tl
+      | `Bytes _ :: tl ->
+          let () = Printf.printf "`Bytes _ :: tl found\n" in
+          parse tl
+      | `Value _ :: tl ->
+          let () = Printf.printf "`Value _ :: tl found\n" in
+          parse tl
+      | _ ->
+          raise (Failure "unaccounted for case")
+    in
+    let symbols = parse tokens in
+    let oid =
+      try
+        List.find (function `Oid x -> true | _ -> false) symbols
+      with Not_found -> `Oid (Unknown (Asn.OID.of_string "0.0.0.0.0.0.0.0.0.0"))
```

Comment:
Instead of using a "dummy" OID, what do you think about making that type an option?